plugin/ruby_hl_lvar.vim

```diff
@@ -14,31 +14,27 @@ let g:ruby_hl_lvar_show_warnings =
 
 augroup ruby_hl_lvar
 	autocmd!
-	autocmd! Filetype * call Ruby_hl_lvar_filetype()
+	autocmd Filetype ruby call Ruby_hl_lvar_filetype()
+	autocmd BufWinLeave * call ruby_hl_lvar#redraw()
+	autocmd WinEnter    * call ruby_hl_lvar#redraw()
+	autocmd WinLeave    * call ruby_hl_lvar#redraw()
+	autocmd TabEnter    * call ruby_hl_lvar#redraw()
+	autocmd TabLeave    * call ruby_hl_lvar#redraw()
 augroup END
 
-autocmd BufWinEnter * call ruby_hl_lvar#redraw()
-autocmd BufWinLeave * call ruby_hl_lvar#redraw()
-autocmd WinEnter    * call ruby_hl_lvar#redraw()
-autocmd WinLeave    * call ruby_hl_lvar#redraw()
-autocmd TabEnter    * call ruby_hl_lvar#redraw()
-autocmd TabLeave    * call ruby_hl_lvar#redraw()
-
 function! Ruby_hl_lvar_filetype()
-	let groupname = 'vim_hl_lvar_'.bufnr('%')
-	execute 'augroup '.groupname
-		autocmd!
-		if &filetype =~# '\<ruby\>'
-			if g:ruby_hl_lvar_auto_enable
-				call ruby_hl_lvar#refresh(1)
-				autocmd TextChanged <buffer> call ruby_hl_lvar#refresh(0)
-				autocmd InsertEnter <buffer> call ruby_hl_lvar#disable(0)
-				autocmd InsertLeave <buffer> call ruby_hl_lvar#refresh(0)
-			else
-				call ruby_hl_lvar#disable(1)
-			endif
-		endif
+	if !g:ruby_hl_lvar_auto_enable
+		return ruby_hl_lvar#disable(1)
+	endif
+
+	call ruby_hl_lvar#refresh(1)
+	augroup ruby_hl_lvar
+		autocmd! * <buffer>
+		autocmd TextChanged <buffer> call ruby_hl_lvar#refresh(0)
+		autocmd InsertEnter <buffer> call ruby_hl_lvar#disable(0)
+		autocmd InsertLeave <buffer> call ruby_hl_lvar#refresh(0)
 	augroup END
+	return
```

Comment:
Why is the `return` necessary? It looks unnecessary.

Reply:
Thanks, removed it.